pkg/client/namespaced_client_test.go

```diff
@@ -0,0 +1,597 @@
+				},
+				Template: corev1.PodTemplateSpec{
+					ObjectMeta: metav1.ObjectMeta{Labels: map[string]string{"foo": "bar"}},
+					Spec:       corev1.PodSpec{Containers: []corev1.Container{{Name: "nginx", Image: "nginx"}}},
+				},
+			},
+		}
+	})
+
+	Describe("Get", func() {
+
+		BeforeEach(func() {
+			var err error
+			dep, err = clientset.AppsV1().Deployments(ns).Create(ctx, dep, metav1.CreateOptions{})
+			Expect(err).NotTo(HaveOccurred())
+		})
+
+		AfterEach(func() {
+			deleteDeployment(ctx, dep, ns)
+		})
+		It("should successfully Get a namespace-scoped object", func() {
+			name := types.NamespacedName{Name: dep.Name}
+			result := &appsv1.Deployment{}
+
+			Expect(getClient().Get(ctx, name, result)).NotTo(HaveOccurred())
+			Expect(result).To(BeEquivalentTo(dep))
+		})
+
+		It("should get the objects from namespace specified in the client", func() {
+			name := types.NamespacedName{Name: dep.Name, Namespace: "non-default"}
+			result := &appsv1.Deployment{}
+
+			Expect(getClient().Get(ctx, name, result)).NotTo(HaveOccurred())
+			Expect(result).To(BeEquivalentTo(dep))
+		})
+	})
+
+	Describe("List", func() {
+		BeforeEach(func() {
+			var err error
+			dep, err = clientset.AppsV1().Deployments(ns).Create(ctx, dep, metav1.CreateOptions{})
+			Expect(err).NotTo(HaveOccurred())
+		})
+
+		AfterEach(func() {
+			deleteDeployment(ctx, dep, ns)
+		})
+
+		It("should successfully List objects when namespace is not specified with the object", func() {
+			result := &appsv1.DeploymentList{}
+			opts := client.MatchingLabels(dep.Labels)
+
+			Expect(getClient().List(ctx, result, opts)).NotTo(HaveOccurred())
+			Expect(len(result.Items)).To(BeEquivalentTo(1))
+			Expect(result.Items[0]).To(BeEquivalentTo(*dep))
+		})
+
+		It("should List objects from the namespace specified in the client", func() {
+			result := &appsv1.DeploymentList{}
+			opts := client.InNamespace("non-default")
+
+			Expect(getClient().List(ctx, result, opts)).NotTo(HaveOccurred())
+			Expect(len(result.Items)).To(BeEquivalentTo(1))
+			Expect(result.Items[0]).To(BeEquivalentTo(*dep))
+		})
+	})
+
+	Describe("Create", func() {
+		AfterEach(func() {
+			deleteDeployment(ctx, dep, ns)
+		})
+
+		It("should successfully create object in the right namespace", func() {
+			By("creating the object initially")
+			err := getClient().Create(ctx, dep)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("checking if the object was created in the right namespace")
+			res, err := clientset.AppsV1().Deployments(ns).Get(ctx, dep.Name, metav1.GetOptions{})
+			Expect(err).NotTo(HaveOccurred())
+			Expect(res.GetNamespace()).To(BeEquivalentTo(ns))
+		})
+
+		It("should not create object if the namespace of the object is different", func() {
+			By("creating the object initially")
+			dep.SetNamespace("non-default")
+			err := getClient().Create(ctx, dep)
+			Expect(err).To(HaveOccurred())
+		})
+
+		It("should create an object in the namespace specified with the client", func() {
```

Comment:
This test is identical to `It("should successfully create object in the right namespace"` or not?